### DISCUSSION TITLE
Trigger js function on livewire element update

### DISCUSSION THREAD

**Discussion by adrianaditech** [livewire/livewire#8276] [1. Help]
In my livewire view I have this textarea
`<textarea wire:model.lazy="note" id="noteEntry" class="form-control auto-resize" placeholder="Write your note here..." required autofocus></textarea>
`

I have this in my layout blade:
```
<script>
    // auto resize textarea
    function resizeTextarea(textarea) {
        textarea.style.height = "auto";
        textarea.style.height = textarea.scrollHeight + "px";
    }

    function setupAutoResize() {
        // Attach the event listener to all textareas with the "auto-resize" class
        document.querySelectorAll(".auto-resize").forEach(function(textarea) {
            textarea.addEventListener("input", function() {
                resizeTextarea(this);
            });

            // Trigger initial resize
            resizeTextarea(textarea);
        });
    }

    // Run the setupAutoResize function when the DOM content is loaded
    document.addEventListener("DOMContentLoaded", function() {
        setupAutoResize();
    });
</script>
<style>
    .auto-resize {
        resize: none;
        overflow: hidden;
    }
</style>
```
It makes all textareas automatically resize to fit contents which works fine if the textarea is rendered with the component the first time. however once livewire sets contents of a textarea (`$this->note = "foo"`), it does not resize it automatically again.

However, when I want to edit an existing note, for example by pressing this
`<button wire:click="editNote({{ $note->id }})">Edit</button>
`
the textarea does not resize.

I tried adding this to the js

```
Livewire.hook('element.updated', (el, component) => {
                if (el.id === 'noteEntry') {
                    window.resizeTextarea(el);
                }
            });
```
but it did not work.
How can I add a livewire listener the js to trigger the resizeTextarea?

**Comment by adrianaditech** [comment]
any suggestions

**Reply by onemoreahmad** [reply]
Try wire:ignore to the element holds the textarea

**Comment by morceaudebois** [comment]
Do you remember if you've had any luck on this? I'm having a similar problem

**Reply by adrianaditech** [reply]
No, I’m still having the same issue

**Reply by morceaudebois** [reply]
I just found a workaround for my problem 😊

I think what I'd try for yours is to make a Livewire component for the textareas. This way you can have HTML + script at the same place. Then what you can do is place your resizing logic in a global function, like this: 

```php
@script
    <script>
        window.resizeTextArea = (textarea) => {
          textarea.style.height = "auto"
          textarea.style.height = textarea.scrollHeight + "px"
        }
    </script>
@endscript
```

Then you will be able to call your function on the textarea using Alpine, like this:

```php
<textarea x-on:input="resizeTextArea($el)" wire:model.lazy="note" id="noteEntry" class="form-control auto-resize" placeholder="Write your note here..." required autofocus></textarea>
```

This will call the function on each matching textarea whenever you type in it. And if I understand your problem correctly, I think it should work (or at least be a good lead).

**Reply by adrianaditech** [reply]
@morceaudebois  thank you very much for your reply.

I already had the auto resize working for when you type in the textarea (then it automatically resizes).
My problem was that the resizeTextArea is not getting triggered for the livewire textarea automatically on page load or when the livewire model's text content is updated from the class.

> It makes all textareas automatically resize to fit contents which works fine if the textarea is rendered with the component the first time. however once livewire sets contents of a textarea ($this->note = "foo"), it does not resize it automatically again.

I have simply this textarea:
`<textarea wire:model.lazy="note" id="noteEntry" class="form-control auto-resize" placeholder="Write your note here..." required autofocus></textarea>
`
And this script:

```
@script
        <script>
            // Define the resizeTextarea function globally
            window.resizeTextarea = function(textarea) {
                textarea.style.height = "auto";
                textarea.style.height = textarea.scrollHeight + "px";
            };

            // Define the setupAutoResize function globally
            window.setupAutoResize = function() {
                // Attach the event listener to all textareas with the "auto-resize" class
                document.querySelectorAll(".auto-resize, .autoresize-allowresize").forEach(function(textarea) {
                    textarea.addEventListener("input", function() {
                        window.resizeTextarea(this); // Reference the global function
                    });

                    // Trigger initial resize
                    window.resizeTextarea(textarea); // Reference the global function
                });
            };

            // Run the setupAutoResize function when the DOM content is loaded
            document.addEventListener("DOMContentLoaded", function() {
                window.setupAutoResize(); // Call the global function
            });
        </script>
@endscript
```

When you type in the textarea, it does auto resize correctly (no need for fancy alpine etc. as this works the same).
However, the window.setupAutoResize does not do anything in the livewire view. When I use it in regular blade views, it works perfectly fine but not in Livewire.
